content/posts/2019/06/xunit-unity-runner/index.eng.md

```diff
@@ -0,0 +1,191 @@
+part of the app and then runs it. 
+
+At first, we thought we needed to build on all three operating systems, but fortunately, Unity supported [cross-compiling][6] 
+and we were able to build the app for MacOS and Windows on Linux.
+
+{{< 
+figure 
+src="github-actions.png" 
+width="823" 
+caption="Being Built on GitHub Actions"
+>}}
+
+Conclusion
+----------
+
+{{< 
+figure 
+src="after.png" 
+width="298" 
+caption="Current Build with Unit Test Added in Unity Environment"
+>}}
+
+The newly built xUnit.net test runner for Unity has been applied on Libplanet project and is currently working well. 
+By working well, we mean that the tests are often breaking due to
+different actions that are only seen in Unity environments. 😇 Of course, we're glad to accept it because that's the 
+point of building a unit test-- to find bugs as early as possible.
+
+The runner is not yet neatly organized, but we have still [put it up as an open source on GitHub][7]. The executable 
+file is available on the [release page][8], so if you want to try it out, it's all yours!
+
+[Libplanet]: https://libplanet.io/
+[regressions]: https://en.wikipedia.org/wiki/Software_regression
+[Azure Pipelines]: https://dev.azure.com/planetarium/libplanet/_build?definitionId=3
+[Travis CI]: https://travis-ci.com/
+[unity-mono]: https://github.com/Unity-Technologies/mono
+[ZeroMQ]: http://zeromq.org/
+[NetMQ]: https://github.com/zeromq/netmq
+[NUnit]: https://nunit.org/
+[xUnit.net]: https://xunit.net/
+[xmldoc]: https://docs.microsoft.com/dotnet/csharp/programming-guide/xmldoc/
+[IoC]: https://en.wikipedia.org/wiki/Inversion_of_control
+[IMessageSinkWithTypes]: https://github.com/xunit/xunit/blob/2.4.1/src/xunit.runner.utility/Messages/IMessageSinkWithTypes.cs
+[Debug.Log]: https://docs.unity3d.com/ScriptReference/Debug.Log.html
+[stdout]: https://en.wikipedia.org/wiki/Standard_streams#Standard_output_(stdout)
+[Console]: https://docs.microsoft.com/dotnet/api/system.console
+[GetCommandLineArgs]: https://docs.microsoft.com/dotnet/api/system.environment.getcommandlineargs
+[Quit]: https://docs.unity3d.com/ScriptReference/Application.Quit.html
+[GitHub Actions]: https://github.com/features/actions
+[5]: https://medium.com/@neuecc/using-circle-ci-to-build-test-make-unitypackage-on-unity-9f9fa2b3adfd
+[6]: https://en.wikipedia.org/wiki/Cross_compiler
+[7]: https://github.com/planetarium/xunit-unity-runner
+[8]: https://github.com/planetarium/xunit-unity-runner/releases
+
+
+[^1]: As of June 2019, <abbr title= “continuous integration”>CI</abbr> services that support all Linux, MacOS, and 
+Windows include [Travis CI] and Azure Pipelines. Our team used Travis CI at first, but it didn't perform well, so 
+we're now using Azure Pipelines. 
+[^2]: Because the .NET Framework supports only Windows, it will be tested in 7 environments instead of 9. 
+[^3]: Because the .NET <abbr title="integrated development environment">IDE</abbr> has become very common for quite 
```

Comment:
```suggestion
[^3]: Because the .NET <abbr title="integrated development environment">IDE</abbr> has become very common for quite some time, there are many projects that don't post API documents on the Web and simply leave [XML document annotations][xmldoc] in the source code. Those annotations will appear small as a tooltip when the class or method is automatically completed in IDE.
```